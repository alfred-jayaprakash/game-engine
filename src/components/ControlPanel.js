import React from 'react';
import {Row, Col} from 'reactstrap';

const ControlPanel = props => {
  return <Row><Col><h4>Control Panel</h4></Col></Row>;
};

export default ControlPanel;
