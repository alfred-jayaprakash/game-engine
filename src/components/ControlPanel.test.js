import React from 'react';
import {render, fireEvent, screen, act, waitFor} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {shallow} from 'enzyme';
import ControlPanel from './ControlPanel';

let container;

describe ('Snapshot tests', () => {
  beforeAll (() => {
    //ARRANGE
  });

  it ('should render correctly without issues', () => {
    const component = shallow (<ControlPanel />);

    expect (component).toMatchSnapshot ();
  });
});

describe ('Functional tests', () => {
  beforeAll (() => {
    //ARRANGE
  });

  beforeEach (() => {
    container = document.createElement ('div');
    document.body.appendChild (container);
  });

  afterEach (() => {
    document.body.removeChild (container);
    container = null;
  });

  it ('initial screen should load correctly', () => {
    render (<ControlPanel />, container);
  });
});
