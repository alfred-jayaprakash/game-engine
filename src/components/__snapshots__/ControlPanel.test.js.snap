// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests should render correctly without issues 1`] = `
<Row
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <h4>
      Control Panel
    </h4>
  </Col>
</Row>
`;
